tests/test_admin/test_routers.py

```diff
@@ -0,0 +1,98 @@
+from fastapi import status
+from fastapi.testclient import TestClient
+
+from src.apps.user.schemas import UserOutputSchema
+from tests.test_orders.conftest import db_orders
+from tests.test_products.conftest import db_categories, db_products
+from tests.test_users.conftest import (auth_headers, create_superuser, db_user,
+                                       superuser_auth_headers)
+
+
+def test_superuser_can_get_all_superusers(
+    sync_client: TestClient, superuser_auth_headers: dict[str, str]
+):
+    response = sync_client.get("admin/superusers", headers=superuser_auth_headers)
+
+    assert response.status_code == status.HTTP_200_OK
+    assert response.json()["total"] == 1
+
+
+def test_new_superuser_can_get_all_superusers(
+    sync_client: TestClient,
+    superuser_auth_headers: dict[str, str],
+    db_user: UserOutputSchema,
+    auth_headers: dict[str, str],
+):
+    sync_client.patch(
+        f"admin/grant-superuser-permissions/user/{db_user.id}",
+        headers=superuser_auth_headers,
+    )
+
+    response = sync_client.get("admin/superusers", headers=auth_headers)
+
+    assert response.status_code == status.HTTP_200_OK
+    assert response.json()["total"] == 2
+
+
+def test_non_superuser_cannot_get_all_superusers(
+    sync_client: TestClient, auth_headers: dict[str, str]
+):
+    response = sync_client.get("admin/superusers", headers=auth_headers)
+
+    assert response.status_code == status.HTTP_403_FORBIDDEN
+
+
+def test_superuser_can_grant_superuser_status(
+    sync_client: TestClient,
+    superuser_auth_headers: dict[str, str],
+    db_user: UserOutputSchema,
+):
```

Comment:
this functionality needs to be analized if it is safe or just staff role will be enough. maybe superuser should grant staff permissions and should not grant superuser permissions as only one superuser will probably exist